src/chainstate/stacks/db/mod.rs

```diff
@@ -748,6 +751,21 @@ impl StacksChainState {
         Ok(path_str)
     }
 
+    fn parse_genesis_address(addr: &str, address_version: u8) -> PrincipalData {
```

Comment:
I think this function, rather than accepting `address_version`, should take the `mainnet` bool.

Then, the address version needs to differentiate between single sig and multi sig. As currently written, it just forces all addresses to single sig, which will break any of the multi sig addresses.

Reply:
Good catch, fixed!